src/unittest/test_serialization.cpp

```diff
@@ -631,6 +633,35 @@ void TestSerialization::testBufReader()
 	UASSERT(!buf.getRawDataNoEx(raw_data, sizeof(raw_data)));
 }
 
+void TestSerialization::testFloatFormat()
+{
+	u32 i;
+	f32 fs, fm;
+	auto test_single = [&fs, &fm](const u32 &i) -> bool {
```

Comment:
The test in this function compares the output of [fu]32To[uf]32Slow with a known-good IEEE implementation, and relies on int32_t and float having the same size and endianness. As such, it should only be run when `getFloatSerializationType()` returns `FLOATTYPE_SYSTEM`, as it's not prepared to run in the platforms that the generic functions are intended to cover.

One alternative test for the case where it returns `FLOATTYPE_SLOW` would be to check some floats against a range of possible encodings. For example, check that f32Tou32Slow(0.1f), which should be 0x3dcccccd, is e.g. between 0x3dccccc80 and 3dccccccff (lest the precision loss makes it inviable for running MT, e.g. a 17 bits mantissa yields only 1/4 block precision in the range 16384..32768).